File: Python/Sliding Window/2130. Maximum Twin Sum of a Linked List.py
""" 2130. Maximum Twin Sum of a Linked List
Difficulty: Medium
Submission Link: https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-given-length/submissions/1387473525/ """

class Solution(object):
    def maxVowels(self, s, k):
        currWindow, currLen, currVowelCount = [], 0, 0
        left = 0
        n = len(s)
        vowels = "AEIOUaeiou"
        ret = 0

        for right in range(n):
            letter = s[right]
            currWindow.append(letter)
            currLen += 1
            if letter in vowels:
                currVowelCount += 1
            
            while currLen > k:
                removedLetter = s[left]
                currWindow.remove(removedLetter)
                currLen -= 1
                if removedLetter in vowels:
                    currVowelCount -= 1
                left += 1
            if currLen == k:
                ret = max(ret, currVowelCount)
        
        return ret

File: C/Sliding Window/2130. Maximum Twin Sum of a Linked List.c
/* 2130. Maximum Twin Sum of a Linked List
Difficulty: Medium
Submission Link:https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-given-length/submissions/1387473829/ */

int maxVowels(char* s, int k) {
    int max=0;
    for(int i=0;i<k;i++){
        if(s[i]=='a' || s[i]=='e' || s[i]=='i' || s[i]=='o' || s[i]=='u'){
            max++;
        }
    }
    int temp=max;
    int right=k;
    while(s[right]!='\0'){
        if(s[right]=='a' || s[right]=='e' || s[right]=='i' || s[right]=='o' || s[right]=='u'){
            temp++;
        }
        if(s[right-k]=='a' || s[right-k]=='e' || s[right-k]=='i' || s[right-k]=='o' || s[right-k]=='u'){
            temp--;
        }
        if(max<temp){
            max=temp;
        }
        right++;
    }
    return max;
}